deeppavlov/agents/ecommerce_agent/ecommerce_agent.py

```diff
@@ -1,55 +1,214 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import defaultdict
+from typing import List, Dict, Any
 import argparse
 
-from deeppavlov.deep import find_config
+from deeppavlov.core.agent.agent import Agent
+from deeppavlov.core.common.log import get_logger
+from deeppavlov.core.skill.skill import Skill
 from deeppavlov.core.commands.infer import build_model_from_config
-from deeppavlov.core.agent.processor import Processor
-from deeppavlov.agents.default_agent.default_agent import DefaultAgent
 from deeppavlov.core.agent.rich_content import RichMessage
 from deeppavlov.agents.rich_content.default_rich_content import PlainText, ButtonsFrame, Button
+from deeppavlov.deep import find_config
 from utils.ms_bot_framework_utils.server import run_ms_bot_framework_server
 
-
 parser = argparse.ArgumentParser()
-parser.add_argument("-i", "--ms-id", help="microsoft bot framework app id", type=str)
-parser.add_argument("-s", "--ms-secret", help="microsoft bot framework app secret", type=str)
+parser.add_argument(
+    "-i", "--ms-id", help="microsoft bot framework app id", type=str)
+parser.add_argument("-s", "--ms-secret",
+                    help="microsoft bot framework app secret", type=str)
+
+log = get_logger(__name__)
+
+
+class EcommerceAgent(Agent):
+    """DeepPavlov Ecommerce agent.
+
+    Args:
+        skill: List of initiated agent skills instances.
+
+    Attributes:
+        skill: List of initiated agent skills instances.
+        history: Histories for each each dialog with agent indexed
+            by dialog ID. Each history is represented by list of incoming
+            and outcoming replicas of the dialog.
+        states: States for each each dialog with agent indexed by dialog ID.
+    """
+
+    def __init__(self, skills: List[Skill], *args, **kwargs) -> None:
+
+        super(EcommerceAgent, self).__init__(skills=skills)
+        self.history: dict = defaultdict(list)
```

Comment:
`self.history` is set like this in the parent class

Reply:
done